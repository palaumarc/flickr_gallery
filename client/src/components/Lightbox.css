.lightbox {
    position: fixed;
    z-index: 1;
    left: 0;
    top: 0;
    width: 100%;
    height: 100%;
    overflow: auto;
    background-color: rgba(70, 70, 70, 0.7);
    display: grid;
    grid-template-rows: 40px 1fr;
}

.lightbox-content {
    height: 100%;
    width: 100%;
}

/* Modal Close Button */
.lightbox-close {
    color: white;
    font-size: 35px;
    font-weight: bold;
    cursor: pointer;
    z-index: 2;
    justify-self: end;
    align-self: center;

    padding-right: 10px;
}

.lightbox-close:hover,
.lightbox-close:focus {
    color: rgb(212, 212, 212);
    text-decoration: none;
}