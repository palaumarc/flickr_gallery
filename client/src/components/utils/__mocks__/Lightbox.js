import React from "react";

const Lightbox = props => <lightbox-mock {...props}>{props.children}</lightbox-mock>;

export default Lightbox;
