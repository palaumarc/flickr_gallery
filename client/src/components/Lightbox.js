import React from 'react';
import './Lightbox.css'

const Lightbox = ({ children, onClose }) => (
    <div className="lightbox">
        <div className="lightbox-close" onClick={onClose}>&times;</div>
        <div className="lightbox-content" >
            {children}
        </div>
    </div>
);

export default Lightbox;