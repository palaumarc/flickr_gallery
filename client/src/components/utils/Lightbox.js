import React from 'react';
import PropTypes from 'prop-types';

import './Lightbox.css'

const Lightbox = ({ children, onClickCloseButton }) => (
    <div className="lightbox">
        <div className="lightbox-close" onClick={onClickCloseButton}>&times;</div>
        <div className="lightbox-content" >
            {children}
        </div>
    </div>
);

Lightbox.propTypes = {
    onClose: PropTypes.func
}

export default Lightbox;