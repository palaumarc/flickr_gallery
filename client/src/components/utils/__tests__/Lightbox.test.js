import React from "react";
import renderer from "react-test-renderer";
import Lightbox from "../Lightbox";

describe('Component is rendered with two children.', () => {

    const onClickCloseButton = jest.fn();

    const componentToTest = (
        <Lightbox onClickCloseButton={onClickCloseButton}>
            <div>Lightbox child 1</div>
            <div>Lightbox child 2</div>
        </Lightbox>
    );

    test('Snapshot', () => {
        const tree = renderer.create(componentToTest);
        expect(tree.toJSON()).toMatchSnapshot();
    })

    test('Close button is clicke. OnClickCloseButton prop should have been called one time', () => {
        const tree = renderer.create(componentToTest);
        const allDivs = tree.root.findAllByType('div');
        const closeButtonDiv = allDivs.find(div => div.children == 'Ã—');
        closeButtonDiv.props.onClick();

        expect(onClickCloseButton).toHaveBeenCalledTimes(1);
    })

}) 