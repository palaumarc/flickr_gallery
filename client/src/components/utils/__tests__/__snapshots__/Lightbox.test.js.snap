// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component is rendered with two children. Snapshot 1`] = `
<div
  className="lightbox"
>
  <div
    className="lightbox-close"
    onClick={[Function]}
  >
    Ã—
  </div>
  <div
    className="lightbox-content"
  >
    <div>
      Lightbox child 1
    </div>
    <div>
      Lightbox child 2
    </div>
  </div>
</div>
`;
